ql/src/java/org/apache/hadoop/hive/ql/exec/tez/monitoring/TezJobMonitor.java

```diff
@@ -458,6 +460,18 @@ private void printSummary(boolean success, Map<String, Progress> progressMap) {
     }
   }
 
+  public void endSummary() {
+    console.endSummary();
+  }
+
+  /**
+   * Collecting execution summary might be a bit expensive, only do it if needed for query history.
+   */
+  private boolean shouldCollectSummaryString() {
+    return hiveConf.getBoolVar(HiveConf.ConfVars.HIVE_QUERY_HISTORY_SERVICE_ENABLED) &&
```

Comment:
can we eagerly initialize var in TezJobMonitor constructor instead of evaluation from conf on every method call?

Reply:
yeah